Cleanup various deprecated code and options (#1047)
getsentry/sentry-php: UPGRADE-3.0.md

- The signature of the `Event::getId()` method changed to return an instance of the `EventId` class instead of a `string`
- The signature of the `Options::getDsn()` method changed to always return an instance of the `Dsn` class instead of a `string`
- Removed the `Options::getProjectId`, `Options::getPublicKey` and `Options::getSecretKey` methods, use `Options::getDsn()` instead.
- Removed the `Breadcrumb::LEVEL_CRITICAL` constant. Use `Breadcrumb::LEVEL_FATAL` instead
- Removed the `Breadcrumb::levelFromErrorException()` method

> - Removedc the `PluggableHttpClientFactory` class

Reviewer: ```suggestion
- Removed the `PluggableHttpClientFactory` class
```

> - Removed the `ClientBuilderInterface::setUriFactory()`, `ClientBuilderInterface::setMessageFactory()`, `ClientBuilderInterface::setTransport()`, `ClientBuilderInterface::setHttpClient()`, `ClientBuilderInterface::addHttpClientPlugin()` and `ClientBuilderInterface::removeHttpClientPlugin()` methods. Use `ClientBuilderInterface::setTransportFactory()` instead

Reviewer: Can we rephrase this (and the others) with multiple methods like this? It's should be easier to scan:
```suggestion
- Removed the following methods from `ClientBuilderInterface`, use `ClientBuilderInterface::setTransportFactory()` instead: 
   - `ClientBuilderInterface::setUriFactory()`
   - `ClientBuilderInterface::setMessageFactory()`
   - `ClientBuilderInterface::setTransport()`
   - `ClientBuilderInterface::setHttpClient()`
   - `ClientBuilderInterface::addHttpClientPlugin()`
   - `ClientBuilderInterface::removeHttpClientPlugin()`
```

- The signature of the `Event::getId()` method changed to return an instance of the `EventId` class instead of a `string`
- The signature of the `Options::getDsn()` method changed to always return an instance of the `Dsn` class instead of a `string`

> - Removed the `Options::getProjectId`, `Options::getPublicKey` and `Options::getSecretKey` methods, use `Options::getDsn()` instead.

Reviewer 2: ```suggestion
- Removed the `Options::getProjectId`, `Options::getPublicKey` and `Options::getSecretKey` methods, use `Options::getDsn()` instead
```

> - Removed the `Breadcrumb::LEVEL_CRITICAL` constant. Use `Breadcrumb::LEVEL_FATAL` instead

Reviewer 2: ```suggestion
- Removed the `Breadcrumb::LEVEL_CRITICAL` constant, use `Breadcrumb::LEVEL_FATAL` instead
```

- Removed the `Breadcrumb::levelFromErrorException()` method
- Removed the `PluggableHttpClientFactory` class
- Removed the following methods from the `ClientBuilderInterface` interface, use `ClientBuilderInterface::setTransportFactory()` instead:
  - `setUriFactory()`
  - `setMessageFactory()`
  - `setTransport()`
  - `setHttpClient()`
  - `addHttpClientPlugin()`
  - `removeHttpClientPlugin()`.
- Removed the following methods from the `Options` class, use the `IgnoreErrorsIntegration` integration instead:
  - `getExcludedExceptions()`
  - `Options::setExcludedExceptions()`
  - `Options::isExcludedException()`
  - `Options::getProjectRoot()`
  - `Options::setProjectRoot()`
- Removed the `Context::CONTEXT_USER`, `Context::CONTEXT_RUNTIME`, `Context::CONTEXT_TAGS`, `Context::CONTEXT_EXTRA`, `Context::CONTEXT_SERVER_OS` constants

> - The signature of the `Scope::setUser()` method changed to not accept anymore the `$merge` parameter

Reviewer 2: ```suggestion
- The signature of the `Scope::setUser()` method changed to not accept the `$merge` parameter anymore
```
